### PR TITLE
[release-0.2] Updating versions, common files and dependecies

### PR DESCRIPTION
**Pull request by FilipB**
* versions.yaml was updated so it contains versions from 0.1 release + the latest for 1.22, 1.23, 1.24
* common files were updated from release-1.24
* dependencies were updated from release-1.24
* istio modules in go.mod were set to v1.24.1

### CODE REVIEW

**Review by luksa — COMMENTED** [review]
-You should also update `go.mod` so it uses Istio 1.24.0.-

Nevermind. GitHub was acting up and wasn't displaying the go.mod changes.

File: go.mod [hunk 1]
```diff
@@ -25,8 +25,8 @@ require (
 	gomodules.xyz/jsonpatch/v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.16.3
-	istio.io/client-go v1.24.0-alpha.0.0.20241206224607-eea6e6266b54
-	istio.io/istio v0.0.0-20241208112209-a944481c0eac
+	istio.io/client-go v1.24.1
+	istio.io/istio v0.0.0-20241120181000-5c178358f9c6
```
Comment (luksa) [inline go.mod:29]: Cant't this point to v1.24.1? The tag exists and points to the same commit (5c178358f9c6). 

```suggestion
	istio.io/istio v1.24.1
```

Comment (FilipB) [reply go.mod:29]: That fails with:
```
go: github.com/istio-ecosystem/sail-operator/controllers/istio imports
	istio.io/istio/pkg/ptr: istio.io/istio@v1.24.1: reading https://proxy.golang.org/istio.io/istio/@v/v1.24.1.zip: 404 Not Found
	server response: not found: istio.io/istio@v1.24.1: invalid version: unknown revision v1.24.1
go: github.com/istio-ecosystem/sail-operator/pkg/istiovalues imports
	istio.io/istio/pkg/util/sets: istio.io/istio@v1.24.1: reading https://proxy.golang.org/istio.io/istio/@v/v1.24.1.zip: 404 Not Found
	server response: not found: istio.io/istio@v1.24.1: invalid version: unknown revision v1.24.1
```

Comment (asmigala) [reply go.mod:27]: https://github.com/istio/istio/issues/41702 this is the reason

**Review by dgn — APPROVED** [review]

**Pull request merged**